homeworks/hw1/lisa_simakova/hw1.py

```diff
@@ -0,0 +1,146 @@
+import argparse
+import numpy as np
+import pandas as pd
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from sklearn.metrics import accuracy_score, confusion_matrix
+from torch.utils.data import DataLoader
+from sklearn.preprocessing import StandardScaler
+import logging
+from torch.utils.data import TensorDataset, DataLoader
+import os
+
+
+logging.basicConfig(
+    filename='training_tmp.log',  
+    filemode='w',
+    format='%(asctime)s - %(levelname)s - %(message)s',
+    level=logging.INFO
+)
+console = logging.StreamHandler()
+console.setLevel(logging.INFO)
+formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+console.setFormatter(formatter)
+logging.getLogger().addHandler(console)
+
+
+def load_data(train_csv, val_csv, test_csv):
+    scaler = StandardScaler()
+    train_df = pd.read_csv(train_csv).drop(columns=['order1', 'order2'])
+    train_df_np = scaler.fit_transform(train_df.iloc[:, :-1])
+    X_train = torch.tensor(train_df_np, dtype=torch.float32)
+    y_train = torch.tensor(train_df.iloc[:, -1].values, dtype=torch.long)
+
+    val_df = pd.read_csv(val_csv).drop(columns=['order1', 'order2'])
+    val_df_np = scaler.transform(val_df.iloc[:, :-1])
+    X_val = torch.tensor(val_df_np, dtype=torch.float32)
+    y_val = torch.tensor(val_df.iloc[:, -1].values, dtype=torch.long)
+
+    test_df = pd.read_csv(test_csv)
+    test_df = scaler.transform(test_df)
+    X_test = torch.tensor(test_df, dtype=torch.float32)
+    return X_train, y_train, X_val, y_val, X_test
+
+
+def init_model(lr=None):
+    class MLP(nn.Module):
+        def __init__(self):
+            super(MLP, self).__init__()
+            self.fc1 = nn.Linear(360, 128)
+            self.dropout1 = nn.Dropout(0.3)
+            self.fc2 = nn.Linear(128, 64)
+            self.dropout2 = nn.Dropout(0.3)
+            self.fc3 = nn.Linear(64, 3)
+
+        def forward(self, x):
+            x = torch.relu(self.fc1(x))
+            x = self.dropout1(x)
+            x = torch.relu(self.fc2(x))
+            x = self.dropout2(x)
+            x = self.fc3(x)
+            return x
```

Comment:
Declaring a class inside a function without explicit necessity is not a good idea